src/main/java/housing/Household.java

```diff
@@ -127,24 +198,55 @@ public void step() {
+        Region targetRegion=region;
+        boolean toTry=decideToTryNeighbouringRegion(geography,region);//to try its neighbouring region or not
+        if(toTry){
+        	targetRegion=selectNewCandidateRegionToBid(region,geography);// select a neighbouring region at random
+        }        
+        
         if(isInSocialHousing()) {
-            bidForAHome(region); // When BTL households are born, they enter here the first time!
+            bidForAHome(targetRegion,transport); // When BTL households are born, they enter here the first time!
         } else if(isRenting()) {
             if(housePayments.get(home).nPayments == 0) { // end of rental period for renter
                 endTenancy();
-                bidForAHome(region);
+                bidForAHome(targetRegion,transport);//TODO: Tony: not sure what happens if these renters fail to get any houses? will they become homeless?
             }            
         } else if(behaviour.isPropertyInvestor()) {
             // TODO: This needs to be broken up in two "decisions" (methods), one for quickly disqualifying investors
             // TODO: who can't afford investing, and another one that, running through the regions, decides whether to
             // TODO: invest there or not (decideToBuyToLetInRegion). How to choose between regions in unbiased manner?
-            if(behaviour.decideToBuyBuyToLet(this, region)) {
-                region.houseSaleMarket.BTLbid(this, behaviour.btlPurchaseBid(this, region));
+            if(behaviour.decideToBuyBuyToLet(this, targetRegion)) {
+            	targetRegion.houseSaleMarket.BTLbid(this, behaviour.btlPurchaseBid(this, targetRegion));
             }
         } else if (!isHomeowner()){
             System.out.println("Strange: this household is not a type I recognize");
         }
     }
+    
+    /**
+     * it is assumed that households may try to bid in their neighbouring regions when they fail to get houses for a certain 
+     * number of steps (or months): maxNumOfStepsInPresentRegion  TODO: this parameter needs to be specified in the config file. 
+     * Currently, maxNumOfStepsInPresentRegion is fixed as 10, only for test purpose
+     * Each of these housholds is further assumed to choose a neighbouring region at radome
+     */
+    public boolean decideToTryNeighbouringRegion(ArrayList<Region> geography,Region presentRegion){  
+    	int maxNumOfStepsInPresentRegion=10;//TODO to be specfied in the config file.
+    	boolean toTry=false;//to try its neighbouringRegion
+    	if(geography.size()>1){// there are other optional regions, apart from the current one.
+    		toTry=presentRegion.decideToTryNeighbouringRegion(householdId, maxNumOfStepsInPresentRegion);    		
+    	}    	
+    	return toTry;
+    }
+    
+    public Region selectNewCandidateRegionToBid(Region presentRegion,ArrayList<Region> geography){ 
```

Comment:
Yes, this random selection method can be improved by selecting among a list of candidate regions. These candidates can be generated based on different rules considering travel time and fees, and average price.  For example, those regions fall outside of the maximum acceptable commuting time (e.g., 1 hour) will not be listed as candidates. Furthermore, this maximum may vary from one household to another. Actually, it could become a complicated question, depending on how precisely we would like to model the behavior.

Reply:
Yes, you're right that this might become complicated if we try to model it in too great detail. See more comments in the email.

Reply:
See my response in the Email.